### PR TITLE
[ServiceBus] Replace get_*_deadletter_receiver functions with a sub_queue parameter

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Remove get_*_deadletter_receiver functions and add a sub_queue parameter to get_*_receiver functions taking enum SubQueue to specify the target subqueue.

Adjusts tests, docs accordingly.

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * Sending a message twice will no longer result in a MessageAlreadySettled exception.
 * `ServiceBusClient.close()` now closes spawned senders and receivers.
 * Attempting to initialize a sender or receiver with a different connection string entity and specified entity (e.g. `queue_name`) will result in an AuthenticationError
+* `get_dead_letter_[queue,subscription]_receiver()` has been removed.  To connect to a dead letter queue, utilize the `SubQueue` parameter of `get_[queue,subscription]_receiver()` provided with a value from the `SubQueue` enum
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:14]:
```suggestion
* `get_dead_letter_[queue,subscription]_receiver()` has been removed.  To connect to a dead letter queue, utilize the `sub_queue` parameter of `get_[queue,subscription]_receiver()` provided with a value from the `SubQueue` enum
```

Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/CHANGELOG.md:14]: NOT BLOCKING: `SubQueue` doesn't sound like a good name for enum.. probably `SubQueueType`?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py [hunk 2]
```diff
@@ -115,6 +115,11 @@ class SessionFilter(Enum):
     NextAvailable = 0
 
 
+class SubQueue(Enum):
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:118]: Should it be named as `SubQueueType`?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py [hunk 3]
```diff
@@ -115,6 +115,11 @@ class SessionFilter(Enum):
...
+    DeadLetter = 1
+    TransferDeadLetter = 2
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:120]: I have some thoughts on the naming, but we could discuss in the next sprint.

like:
```python
class SubQueueType(Enum):
    DeadLetterQueue = 1
    TransferDeadLetterQueue = 2
```

Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:120]: See the poll that Ramya ran on this exact question :)  (I voted for SubQueueType, for the record :P )

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py [hunk 4]
```diff
@@ -192,6 +193,9 @@ def get_queue_receiver(self, queue_name, **kwargs):
         """Get ServiceBusReceiver for the specific queue.
 
         :param str queue_name: The path of specific Service Bus Queue the client connects to.
+        :keyword Optional[SubQueue] sub_queue: If specified, the subqueue this receiver will connect to.
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:196]: nit: do we want the docstring in this format:
```python
:keyword sub_queue:
:paramtype sub_queue: Optional[~azure.servicebus.SubQueue]
```

File: sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py (hunk 3 above)
Comment (yunhaoling) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py:120]: lol, we're the minority.
well.. I can live with it. not a big issue

**Review by yunhaoling — APPROVED** [review]

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py (hunk 4 above)
Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:196]: Does it change the doc rendering in any meaningful way?  I can check with sphinx but I'm inclined to make this a note on one of our docstring cleanup bugs for next sprint.  (am still curious what tractable difference it'd make to the reference)

Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py:196]: (Added to docstring cleanup issue a note to this discussion and steps to investigate impact)

**Review by yunhaoling — DISMISSED** [review]

**Review by yunhaoling — APPROVED** [review]

**Review by lmazuel — APPROVED** [review]

**Pull request merged**